lineapy/plugins/serializers/mlflow_io.py

```diff
@@ -0,0 +1,170 @@
+            "deserializer": mlflow.sklearn.load_model,
+        }
+    except Exception:
+        pass
+
+    try:
+        from prophet import Prophet
+
+        mlflow_io["prophet"] = {
+            "class": [Prophet],
+            "serializer": mlflow.prophet.log_model,
+            "deserializer": mlflow.prophet.load_model,
+        }
+    except Exception:
+        pass
+
+    try:
+        from xgboost.core import Booster
+
+        mlflow_io["xgboost"] = {
+            "class": [Booster],
+            "serializer": mlflow.xgboost.log_model,
+            "deserializer": mlflow.xgboost.load_model,
+        }
+    except Exception:
+        pass
+
+    try:
+        from statsmodels.base.wrapper import ResultsWrapper
+
+        mlflow_io["statsmodels"] = {
+            "class": [ResultsWrapper],
+            "serializer": mlflow.statsmodels.log_model,
+            "deserializer": mlflow.statsmodels.load_model,
+        }
+    except Exception:
+        pass
+
+except ImportError:
+    pass
+
+
+def try_write_to_mlflow(value: Any, name: str, **kwargs) -> Optional[Any]:
+    """
+    Try to save artifact with MLflow
+
+    Parameters
+    ----------
+    value: Any
+        value(ML model) to save with mlflow
+    name: str
+        artifact_path and registered_model_name used in
+        `mlflow.sklearn.log_model` or equivalent flavors
+    **kwargs:
+        args to pass into `mlflow.sklearn.log_model` or equivalent flavors
+
+    Returns
+    -------
+    Optional[Any]
+        return a ModelInfo(MLflow model metadata) if successfully save with
+        mlflow; otherwise None.
+
+    Note that, using Any for type checking in case mlflow is not installed.
+
+    """
+
+    logger.info("Trying to save the object to MLflow.")
+
+    # Check mlflow is installed, if not raise error
+    if "mlflow" not in sys.modules:
+        msg = (
+            "module 'mlflow' is not installed;"
+            + " please install it with 'pip install lineapy[mlflow]'"
+        )
+        raise ModuleNotFoundError(msg)
+    mlflow.set_tracking_uri(options.get("mlflow_tracking_uri"))
+
+    # Check value is from a module supported by mlflow
+    full_module_name = getmodule(value)
+    if full_module_name is not None:
+        root_module_name = full_module_name.__name__.split(".")[0]
+        if root_module_name in mlflow_io.keys():
+            flavor_io = mlflow_io[root_module_name]
+            # Check value is the right class type for the module supported by mlflow
+            if any(
+                [
+                    isinstance(value, target_class)
+                    for target_class in flavor_io["class"]
+                ]
+            ):
+                kwargs["registered_model_name"] = kwargs.get(
+                    "registered_model_name", name
+                )
+                kwargs["artifact_path"] = kwargs.get("artifact_path", name)
+                model_info = flavor_io["serializer"](value, **kwargs)
```

Comment:
Add a code comment here saying this line is the actual "save" 

Its kind of non-obvious since the actual logic is looked up from a dictionary ...